pkg/logql/log/filter.go

```diff
@@ -142,46 +184,46 @@ func newRegexpFilter(re string, match bool) (Filterer, error) {
 	if err != nil {
 		return nil, err
 	}
-	f := regexpFilter{reg}
+	f := RegexpFilter{reg}
 	if match {
 		return f, nil
 	}
 	return newNotFilter(f), nil
 }
 
-func (r regexpFilter) Filter(line []byte) bool {
+func (r RegexpFilter) Filter(line []byte) bool {
 	return r.Match(line)
 }
 
-func (r regexpFilter) ToStage() Stage {
+func (r RegexpFilter) ToStage() Stage {
 	return StageFunc{
 		process: func(line []byte, _ *LabelsBuilder) ([]byte, bool) {
 			return line, r.Filter(line)
 		},
 	}
 }
 
-type containsFilter struct {
+type ContainsFilter struct {
```

Comment:
do you think it's possible to keep all of this private and expose just a way to accumulate Filters in `log` package.

Reply:
We could try to run the logic on the `andFilter`